Balloon/sys/Device.c

```diff
@@ -367,20 +367,26 @@ BalloonEvtDeviceD0Entry(IN WDFDEVICE Device, IN WDF_POWER_DEVICE_STATE PreviousS
     if (!NT_SUCCESS(status))
     {
         TraceEvents(TRACE_LEVEL_ERROR, DBG_PNP, "BalloonInit failed with status 0x%08x\n", status);
-        BalloonTerm(Device);
-        return status;
+        goto Exit;
```

Comment:
For additional clarity maybe it worth renaming BalloonTerm to something like BalloonDeleteQueues?

If the name stays, I would goto Terminate and not Exit on line 370 for future generation clarity, while it functionally is not needed - to verify it is not need is a walk through 3-4 functions :)

Reply:
Thanks!